File: PacifistaPx0-cs50-2018-x-sentimental-mario-more/mario.py
import sys

# mario implemented python


def mario():
    # prompt user for none negative integer and error check using valueError

    while True:
        try:
            height = int(input("What is your height: "))
        except ValueError:
            print('Please enter a valid integer')
            continue
        if height >= 0 and height < 24:
            break
    for i in range(1, height+1):
        j, k, l = 0, 0, 0

        # this will loop print spaces until it is time to print
        for space in range(height-i):
            print(" ", end="")
            space += 1

        # this will loop print #
        while j != i:
            print("#", end="")
            j += 1

        # print the gap between the half pyramid because k will always be less than i
        if k < i:
            print("  ", end="")

        # loop print # after the gaps for the second pyramid
        while l != i:
            print("#", end="")
            l += 1

        # restart the loop on a new line
        print()


mario()

File: PacifistaPx0-cs50-2018-x-mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // all variables subjected to change.
    int i, space, height, j = 0, k = 0, l = 0;
    do
    {
        //prompt user for height
        height = get_int("I require a positive number between 0 and 23 for the height of your pyramid: ");
    }
    while (height < 0 || height > 23);

    //we increment i by 1 and reset j, k and l to 0 whenever this loops
    for (i = 1; i <= height; i++, j = 0, k = 0, l = 0)
    {

        //this will loop print spaces until it is time to print #
        for (space = 1; space <= height - i; space++)
        {
            printf(" ");
        }

        //this will loop print #
        while (j != i)
        {
            printf("#");
            j++;
        }

        //print the gap between the half pyramid because k will always be less than i
        if (k < i)
        {
            printf("  ");
        }
        //loop print # after the gaps for the second pyramid
        while (l != i)
        {
            printf("#");
            l++;
        }

        //restart the loop on a new line
        printf("\n");
    }
}